contracts/lib/TimeLockUpgradeV3.sol

```diff
@@ -0,0 +1,155 @@
+/*
+    Copyright 2020 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.5.7;
+
+import { Ownable } from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+
+/**
+ * @title TimeLockUpgradeV3
+ * @author Set Protocol
+ *
+ * The TimeLockUpgradeV3 contract contains a modifier for handling minimum time period updates
+ *
+ * CHANGELOG:
+ * - Requires that the caller is the owner
+ * - New function to allow deletion of existing timelocks
+ * - Added upgradeData to UpgradeRegistered event
+ */
+contract TimeLockUpgradeV3 is
+    Ownable
```

Comment:
Instead of copying all the state and logic, why not inherit `TimeLockUpgradeV2.sol` and then override the timeLockUpgrade function?